dotnet/src/Microsoft.Agents.AI.DurableTask/AgentRunHandle.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.DurableTask.Client;
+using Microsoft.DurableTask.Client.Entities;
+
+namespace Microsoft.Agents.AI.DurableTask;
+
+/// <summary>
+/// Represents a handle for a running agent request that can be used to retrieve the response.
+/// </summary>
+internal sealed class AgentRunHandle
+{
+    private readonly DurableTaskClient _client;
+
+    internal AgentRunHandle(DurableTaskClient client, AgentSessionId sessionId, string correlationId)
+    {
+        this._client = client;
+        this.SessionId = sessionId;
+        this.CorrelationId = correlationId;
+    }
+
+    /// <summary>
+    /// Gets the correlation ID for this request.
+    /// </summary>
+    public string CorrelationId { get; }
+
+    /// <summary>
+    /// Gets the session ID for this request.
+    /// </summary>
+    public AgentSessionId SessionId { get; }
+
+    /// <summary>
+    /// Reads the agent response for this request by polling the entity state until the response is found.
+    /// Uses an exponential backoff polling strategy with a maximum interval of 1 second.
+    /// </summary>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The agent response corresponding to this request.</returns>
+    /// <exception cref="InvalidOperationException">Thrown when the response is not found after polling.</exception>
+    public async Task<AgentRunResponse> ReadAgentResponseAsync(CancellationToken cancellationToken = default)
+    {
+        TimeSpan pollInterval = TimeSpan.FromMilliseconds(50); // Start with 50ms
+        TimeSpan maxPollInterval = TimeSpan.FromSeconds(1); // Maximum 1 second
```

Comment:
Curious how we arrived at 1s as the max?

Reply:
I figured that most responses shouldn't exceed a handful of seconds, so one second polling should be a good balance. But if it caught your attention, then it's probably not the obvious choice. I can bump it up to 3 seconds. I think we can iterate further (or make it configurable) based on developer feedback.